src/AdvApi32.Desktop/AdvApi32.cs

```diff
@@ -888,14 +933,169 @@ public static partial class AdvApi32
         /// </returns>
         [DllImport(nameof(AdvApi32), SetLastError = true)]
         [return: MarshalAs(UnmanagedType.Bool)]
-        public static unsafe extern bool QueryServiceStatusEx(
+        public static extern bool QueryServiceStatusEx(
             SafeServiceHandle hService,
             SC_STATUS_TYPE InfoLevel,
-            void* lpBuffer,
+            IntPtr lpBuffer,
             int cbBufSize,
             out int pcbBytesNeeded);
 
         /// <summary>
+        /// Registers a function to handle service control requests.
+        /// </summary>
+        /// <param name="lpServiceName">
+        /// The name of the service run by the calling thread. This is the service name that the service control program specified in the CreateService function when creating the service.
+        /// If the service type is SERVICE_WIN32_OWN_PROCESS, the function does not verify that the specified name is valid, because there is only one registered service in the process.
+        /// </param>
+        /// <param name="lpHandlerProc">A reference to the handler function to be registered.</param>
+        /// <returns>If the function succeeds, the return value is a service status handle If the function fails, the return value is zero. </returns>
+        /// <remarks>
+        /// This function has been superseded by the <see cref="RegisterServiceCtrlHandlerEx"/> function.
+        /// A service can use either function, but the new function supports user-defined context data, and the new handler function supports additional extended control codes.
+        /// </remarks>
+        [DllImport(nameof(AdvApi32), CharSet = CharSet.Unicode)]
+        [Obsolete("Use LPHANDLER_FUNCTION_EX with RegisterServiceCtrlHandlerEx instead")]
```

Comment:
Should we obsolete non-ex functions as a general rule ?
I didn't notice it on other functions here, and it might create problems for users that want to port existing code to this lib without receiving warnings.